src/client/components/Comment/Comment.js

```diff
@@ -2,261 +2,308 @@
+        dangerouslySetInnerHTML={props.children}
+      />)
+  }
+  if (props.onoStyle) {
+    return (
+      <span
+        className="comment-body-onostyle"
+        dangerouslySetInnerHTML={props.children}
+      />
+    )
+  }
+  return (
+    <span className="comment-body-onostyle">
+      {props.children}
+    </span>
+  )
 }
 
 class Comment extends Component {
   /* eslint-disable react/sort-comp */
 
   constructor(props: Props) {
     super(props)
+    marked.setOptions({
+      gfm: true,
+      tables: true,
+      breaks: true,
+      pedantic: false,
+      sanitize: true,
+      smartLists: true,
+      smartypants: false,
+    })
+    const rendered = this.renderCommentText(props.comment.text, props)
     this.state = {
       editMode: props.newComment,
       newComment: props.newComment,
-      commentText: '',
+      commentText: props.comment.text,
+      renderedText: rendered,
     }
   }
 
   props: Props
 
+  formatOnoText(text) {
+    let formatted = escapeHTML(text)
+    formatted = formatted.replace(/@([a-z]+)/g, '<b>@$1</b>')
+    formatted = linkifyRevisionHashes(formatted)
+    // TODO add urlification of urls, revision hashes, and cases
+    return formatted
+  }
+
+  formatMarkdown(text) {
+    let formatted = marked(text)
+    formatted = linkifyRevisionHashes(formatted)
+    return formatted
+  }
+
   onCommentEdit = () => {
-    this.setState({ editMode: true })
+    this.setState({ editMode: !this.state.editMode })
+  }
+
+  onCommentDelete = () => {
+    this.props.onCommentDelete()
   }
 
   onCommentCancel = () => {
+    if (this.state.newComment) {
+      this.props.onCommentCancel()
+    }
     this.setState({
       editMode: false,
       newComment: false,
+      commentText: this.props.comment.text,
     })
-    // TODO: unmount component if this is a new comment.
   }
-  // onCommentDelete() {
-  //   // TODO: dispatch reducer action to delete comment
-  // }
 
   onCommentSave = () => {
     this.setState({
       editMode: false,
       newComment: false,
+      renderedText: this.renderCommentText(this.state.commentText, this.props),
     })
-    // TODO: dispatch reducer action to save edited and new comments
+    this.props.onCommentSave(this.comment.id, this.state.commentText)
+    // TODO: Figure out proper API for onCommentSave
   }
 
-  // handleChange(event, value) {
-  //   switch (value) {
-  //     case '1':
-  //       this.setState({
-  //         issue: !this.state.issue,
-  //       })
-  //       return
-  //     case '2':
-  //       this.setState({
-  //         niceToHave: !this.state.niceToHave,
-  //       })
-  //       return
-  //     case '3':
-  //       this.setState({
-  //         codeStyle: !this.state.codeStyle,
-  //       })
-  //       return
-  //     default:
-  //       return
-  //   }
-  // }
-
-  render() {
-    if (!this.props.comment || !this.props.comment.author) {
-      return null
+  renderCommentText(text, { markdown, onoStyle }) {
```

Comment:
Looks like just text formatting rather than rendering. Maybe nitpick, but this return `CommentBody` specific value, which is most likely should be in  `CommentBody`. I would format and transform it there.